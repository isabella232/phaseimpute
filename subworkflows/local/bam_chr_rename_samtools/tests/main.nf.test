nextflow_workflow {

    name "Test Subworkflow BAM_CHR_RENAME_SAMTOOLS"
    script "../main.nf"

    config "./nextflow.config"

    workflow "BAM_CHR_RENAME_SAMTOOLS"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "subworkflows/bam_chr_rename_samtools"
    tag "bam_chr_rename_samtools"

    tag "samtools"
    tag "samtools/reheader"
    tag "samtools/index"
    tag "gawk"

    test("Should remove chr prefix") {
        when {
            workflow {
                """
                input[0] = Channel.fromList([
                    [
                        [id: "test_paired"],
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExist:true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExist:true),
                        "nochr"
                    ],
                    [
                        [id: "test_MT"],
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test_illumina_mt.bam', checkIfExist:true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test_illumina_mt.bam.bai', checkIfExist:true),
                        "nochr"
                    ],
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() },
                { assert snapshot(workflow.out.bam_renamed.collect{
                    bam(it[1]).getHeader().findAll { it.startsWith ("@SQ") }
                    }).match("headernochr")
                }
            )
        }
    }

    test("Should add chr prefix") {
        setup {
            run("BAM_CHR_RENAME_SAMTOOLS", alias: "PREPROCESS") {
                script "../main.nf"
                process {
                    """
                    input[0] = Channel.fromList([
                        [
                            [id: "test_paired"],
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExist:true),
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExist:true),
                            "nochr"
                        ],
                        [
                            [id: "test_MT"],
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test_illumina_mt.bam', checkIfExist:true),
                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test_illumina_mt.bam.bai', checkIfExist:true),
                            "nochr"
                        ],
                    ])
                    """
                }
            }
        }

        when {
            workflow {
                """
                input[0] = PREPROCESS.out.bam_renamed.combine(["chr"])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() },
                { assert snapshot(workflow.out.bam_renamed.collect{
                    bam(it[1]).getHeader().findAll { it.startsWith ("@SQ") }
                    }).match("headerwithchr")
                }
            )
        }
    }
}
