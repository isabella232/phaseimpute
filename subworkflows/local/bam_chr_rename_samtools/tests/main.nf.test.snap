{
    "Should remove chr prefix": {
        "content": [
            [
                "versions.yml:md5,3777f4a1f04f5087333fbe98eee09df9",
                "versions.yml:md5,fc2d8252e26e8b681f2dd73d8b08fdde"
            ],
            [
                [
                    {
                        "id": "test_MT"
                    },
                    "test_MT_withchr.bam",
                    "test_MT_withchr.bam.bai"
                ],
                [
                    {
                        "id": "test_paired"
                    },
                    "test_paired_withchr.bam",
                    "test_paired_withchr.bam.bai"
                ]
            ],
            [
                [
                    "@SQ\tSN:M\tLN:16569"
                ],
                [
                    "@SQ\tSN:MT192765.1\tLN:29829"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-07T13:27:45.777831517"
    },
    "Should add chr prefix": {
        "content": [
            [
                "versions.yml:md5,3777f4a1f04f5087333fbe98eee09df9",
                "versions.yml:md5,fc2d8252e26e8b681f2dd73d8b08fdde"
            ],
            [
                [
                    {
                        "id": "test_MT"
                    },
                    "test_MT_withchr.bam",
                    "test_MT_withchr.bam.bai"
                ],
                [
                    {
                        "id": "test_paired"
                    },
                    "test_paired_withchr.bam",
                    "test_paired_withchr.bam.bai"
                ]
            ],
            [
                [
                    "@SQ\tSN:chrM\tLN:16569"
                ],
                [
                    "@SQ\tSN:chrMT192765.1\tLN:29829"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-07T13:28:15.125413039"
    }
}